examples/kitchen-sink.yaml

```diff
@@ -762,6 +770,22 @@ resource:
       # If omitted, .included attributes are included.
       excluded:
         - process.command_args
+    # Configure resource detectors. See also enabled_detectors_strategy.
+    # Resource detector names are dependent on the SDK language ecosystem. Please consult documentation for each respective language. 
+    # If omitted or null and enabled_detectors_strategy=opt_in, no resource detectors are enabled.
+    detectors:
+      - # Enable the container resource detector.
+        # Note, the key "container" is an example and detector names may vary by SDK language ecosystem.
+        container:
+      - # Enable the host resource detector.
+        # Note, the key "host" is an example and detector names may vary by SDK language ecosystem.
+        host:
+      - # Enable the os resource detector.
+        # Note, the key "os" is an example and detector names may vary by SDK language ecosystem.
+        os:
+      - # Enable the process resource detector.
+        # Note, the key "process" is an example and detector names may vary by SDK language ecosystem.
+        process:
```

Comment:
The lack of a well known list of standard resource detector names is the biggest problem with this design. I'd like to fix that by adding such a list to the [resource specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#detecting-resource-information-from-the-environment), along with general guidance that resource detectors be named according to the root namespace of attributes they typically provide. For example, a resource detector that provides `os.*` attributes should be named `os`, one that provides `process.*` should be named `process`, etc.

Reply:
> For example, a resource detector that provides `os.*` attributes should be named `os`

I would turn it differently:

A resource detector named `os` should provide `os.*` attributes.

This is ok because the name, `os`, is a prefix for attributes.

It should be also possible to have a resource detector named `os_simulated`, `qa`, or `my_fake_detector`, that also provides `os.*` attributes.